kas-fleet-manager/deploy-kas-fleet-manager.sh

```diff
@@ -141,35 +92,71 @@ deploy_kasfleetmanager() {
   create_kas_fleet_manager_namespace
 
   echo "Deploying KAS Fleet Manager Database..."
-  ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/db-template.yml | oc apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
+  ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/db-template.yml | ${OC} apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
   echo "Waiting until KAS Fleet Manager Database is ready..."
   time timeout --foreground 3m bash -c "until ${OC} get pods -n ${KAS_FLEET_MANAGER_NAMESPACE}| grep kas-fleet-manager-db | grep -v deploy | grep -q Running; do echo 'database is not ready yet'; sleep 10; done"
 
+  create_kasfleetmanager_service_account
+  create_kasfleetmanager_pull_credentials
+
   echo "Deploying KAS Fleet Manager K8s Secrets..."
   ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/secrets-template.yml \
-    -p OCM_SERVICE_CLIENT_ID="dummyclient" \
-    -p OCM_SERVICE_CLIENT_SECRET="dummysecret" \
+    -p OCM_SERVICE_CLIENT_ID="" \
+    -p OCM_SERVICE_CLIENT_SECRET="" \
+    -p OCM_SERVICE_TOKEN="${OCM_SERVICE_TOKEN}" \
     -p OBSERVABILITY_CONFIG_ACCESS_TOKEN="${OBSERVABILITY_CONFIG_ACCESS_TOKEN}" \
     -p MAS_SSO_CLIENT_ID="${MAS_SSO_CLIENT_ID}" \
     -p MAS_SSO_CLIENT_SECRET="${MAS_SSO_CLIENT_SECRET}" \
+    -p OSD_IDP_MAS_SSO_CLIENT_ID="${MAS_SSO_CLIENT_ID}" \
+    -p OSD_IDP_MAS_SSO_CLIENT_SECRET="${MAS_SSO_CLIENT_SECRET}" \
     -p MAS_SSO_CRT="${MAS_SSO_CRT}" \
     -p KAFKA_TLS_CERT="${KAFKA_TLS_CERT}" \
     -p KAFKA_TLS_KEY="${KAFKA_TLS_KEY}" \
-    -p DATABASE_HOST="$(${KUBECTL} get service/kas-fleet-manager-db -o jsonpath="{.spec.clusterIP}")" \
+    -p KUBE_CONFIG="$(${OC} config view --minify --raw | base64 -w0)" \
+    -p IMAGE_PULL_DOCKER_CONFIG=$(${OC} get secret ${KAS_FLEET_MANAGER_IMAGE_PULL_SECRET_NAME} -n ${KAS_FLEET_MANAGER_NAMESPACE} -o jsonpath="{.data.\.dockerconfigjson}") \
     | ${OC} apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
 
   echo "Deploying KAS Fleet Manager Envoy ConfigMap..."
   ${OC} apply -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/envoy-config-configmap.yml -n ${KAS_FLEET_MANAGER_NAMESPACE}
 
-  create_kasfleetmanager_service_account
-  create_kasfleetmanager_pull_credentials
-
   echo "Deploying KAS Fleet Manager..."
   OCM_ENV="development"
 
+  SERVICE_PARAMS=${DIR_NAME}/kas-fleet-manager-params.env
+  > ${SERVICE_PARAMS}
+
+  if [ -n "${OCM_SERVICE_TOKEN}" ] ; then
+      PROVIDER_TYPE="ocm"
+      ENABLE_OCM_MOCK="false"
+      CLUSTER_STATUS="cluster_provisioned"
```

Comment:
@machi1990 is it possible to disable the creation of the identity provider? It seems like setting `ENABLE_OCM_MOCK` installs the IDP, which seems to prevent access to the cluster as kubeadmin.

Reply:
> Can this wait for next week?
Absolutely, have a nice weekend!

Reply:
I've created https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/992 can you have a look? @MikeEdgar

Reply:
lgtm, resolving this. thread. Thanks!

Reply:
Thanks. Were you able to verify that enabling/disabling works as well? If so https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/992 can be merged. /cc @JameelB

Reply:
Yes, it worked in both cases. Omitting the param results in Kafka_SRE idp, and setting to false avoids creating Kafka_SRE idp.

Reply:
Thanks @MikeEdgar for the end to end verification.